internal/server/commonpool.go

```diff
@@ -0,0 +1,31 @@
+package server
+
+import (
+	"math"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+//islandDistribute distributes resources to island as requested
+func (s *SOMASServer) islandDistribute(resource shared.Resources, islandID shared.ClientID) {
+	ci := s.gameState.ClientInfos[islandID]
+	ci.Resources = ci.Resources + resource
+	s.gameState.ClientInfos[islandID] = ci
+	s.gameState.CommonPool -= resource
+}
```

Comment:
Where would be most logical to include this method to distribute CP relief to clients?

Reply:
. Currently clients can simply just take from the common pool in IIGOAllocations and if you expect client to just take from the CP in order to get some disaster relief we can add as part of that so that its packaged nicely.

Reply:
See `applyDisasterEffects()` in `endOfTurn()`. To me it makes most sense to implement disaster effects (including relief based on CP) here since this is a sort of asynchronous event that won't happen on a predictable interval (unlike IIGOAllocations). Thoughts?

Reply:
Well from my understanding of the disaster system the common pool is first depleted from in order to mitigate disaster, I guess that acts as a sort of 'relief' and then any agent who wants to take from the common pool for the purposes of surviving a disaster can take it as normal. Potentially we can add a reason as to why the agent is taking from the common pool but that seems low priority

Reply:
Hmm I don't think that's how our team implemented it. It's more along the lines of
- get 'raw' effect of storm
- get proportional effect for each island (relative to others)
- reduce effect of each island using CP resources in proportion to effect/damage felt
- deduct islands resources based on the final CP-mitigated effects (potentially zero deduction if full mitigation)

Would this way of doing it be a problem?

Reply:
I think this way of calculating disaster effects is fine. I think we actually agree, I just kinda muddled my words. In terms of directly giving resources to clients from the common pool let's leave it to just happen at IIGOAllocations